lib/sqlalchemy/orm/session.py

```diff
@@ -1261,6 +1261,9 @@ def _maker_context_manager(self):
             with self.begin():
                 yield self
 
+    def _create_transaction(self, *args, **kwargs):
+        return SessionTransaction(self, *args,**kwargs)
```

Comment:
Usually for these kind of thing a class variable on the class is used.
Here it could be `transaction_cls = SessionTransaction` then it's called as `self.transaction_cls(...)`

Reply:
Sure. Can be done either way.

This seemed more flexible because you could intercept the creation without subclassing the SessionTransaction. Doesn't matter, since the PR doesn't seem likely to be accepted given the design philosphy issues in the rest of the PR response.

Reply:
> Sure. Can be done either way.
> 
> This seemed more flexible because you could intercept the creation without subclassing the SessionTransaction. Doesn't matter, since the PR doesn't seem likely to be accepted given the design philosphy issues in the rest of the PR response.

It's too early to state that.    I would like the session event system to be able to accommodate your use case.  however if there's some compelling case for why they can't work, or why it's too much burden, I'd want to see that.   It's very difficult for us to field these kinds of requests without knowing what it is someone's trying to do; generally we never add anything at all, not even something uncontroversial, without some understanding of the use case, so that we can make sure we make the best change possible with a clear rationale.   There's lots of examples of weird API features we've had to carry along for years, making it difficult to refactor things, because someone wanted some particular thing at some point which seemed easy enough and we just accepted it.  or, someone wanted something simple, we accepted it, then we realize later the area in which they are working is actually much more complicated than what they needed, someone else needs a similar feature but we've already bolted an incompatible API in place.    so we can't really afford to make that mistake these days, hence we request that we get a clear rationale why something is needed.